recipes/capnproto/all/conanfile.py

```diff
@@ -0,0 +1,185 @@
+        if self.settings.compiler == "Visual Studio" and self.options.shared:
+            raise ConanInvalidConfiguration("Cap'n Proto doesn't support shared libraries for Visual Studio")
+
+    def requirements(self):
+        if self.options.with_openssl:
+            self.requires("openssl/1.1.1h")
+        if self.options.with_zlib:
+            self.requires("zlib/1.2.11")
+
+    def build_requirements(self):
+        if self.settings.os != "Windows":
+            self.build_requires("libtool/2.4.6")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["BUILD_TESTING"] = False
+        self._cmake.definitions["EXTERNAL_CAPNP"] = False
+        self._cmake.definitions["CAPNP_LITE"] = False
+        self._cmake.definitions["WITH_OPENSSL"] = self.options.with_openssl
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        args = [
+            "--enable-shared" if self.options.shared else "--disable-shared",
+            "--disable-static" if self.options.shared else "--enable-static",
+            "--with-openssl" if self.options.with_openssl else "--without-openssl",
+            "--with-zlib" if self.options.with_zlib else "--without-zlib",
+            "--enable-reflection"
+        ]
+        self._autotools = AutoToolsBuildEnvironment(self)
+        self._autotools.configure(args=args, configure_dir=os.path.join(self._source_subfolder, "c++"))
+        return self._autotools
+
+    def build(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        if self.settings.os == "Windows":
+            cmake = self._configure_cmake()
+            cmake.build()
+        else:
+            with tools.chdir(os.path.join(self._source_subfolder, "c++")):
+                self.run("{} --install --verbose -Wall".format(tools.get_env("AUTORECONF")))
+            autotools = self._configure_autotools()
+            autotools.make()
+
+    @property
+    def _cmake_folder(self):
+        return os.path.join("lib", "cmake", "CapnProto")
+
+    def package(self):
+        self.copy("LICENSE", dst="licenses", src=self._source_subfolder)
+        if self.settings.os == "Windows":
+            cmake = self._configure_cmake()
+            cmake.install()
+        else:
+            autotools = self._configure_autotools()
+            autotools.install()
+            for la_file in glob.glob(os.path.join(self.package_folder, "lib", "*.la")):
+                os.remove(la_file)
```

Comment:
This also requires a minimum_conan_version at the top of the recipe.

```suggestion
            tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
```

Reply:
I'm a little bit reluctant to raise minimum conan version just for this function (1.29.1)